<html><body>
<style>

body, h1, h2, h3, div, span, p, pre, a {
  margin: 0;
  padding: 0;
  border: 0;
  font-weight: inherit;
  font-style: inherit;
  font-size: 100%;
  font-family: inherit;
  vertical-align: baseline;
}

body {
  font-size: 13px;
  padding: 1em;
}

h1 {
  font-size: 26px;
  margin-bottom: 1em;
}

h2 {
  font-size: 24px;
  margin-bottom: 1em;
}

h3 {
  font-size: 20px;
  margin-bottom: 1em;
  margin-top: 1em;
}

pre, code {
  line-height: 1.5;
  font-family: Monaco, 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Lucida Console', monospace;
}

pre {
  margin-top: 0.5em;
}

h1, h2, h3, p {
  font-family: Arial, sans serif;
}

h1, h2, h3 {
  border-bottom: solid #CCC 1px;
}

.toc_element {
  margin-top: 0.5em;
}

.firstline {
  margin-left: 2 em;
}

.method  {
  margin-top: 1em;
  border: solid 1px #CCC;
  padding: 1em;
  background: #EEE;
}

.details {
  font-weight: bold;
  font-size: 14px;
}

</style>

<h1><a href="genomics_v1.html">Genomics API</a> . <a href="genomics_v1.annotations.html">annotations</a></h1>
<h2>Instance Methods</h2>
<p class="toc_element">
  <code><a href="#batchCreate">batchCreate(body, x__xgafv=None)</a></code></p>
<p class="firstline">Creates one or more new annotations atomically. All annotations must</p>
<p class="toc_element">
  <code><a href="#create">create(body, x__xgafv=None)</a></code></p>
<p class="firstline">Creates a new annotation. Caller must have WRITE permission</p>
<p class="toc_element">
  <code><a href="#delete">delete(annotationId, x__xgafv=None)</a></code></p>
<p class="firstline">Deletes an annotation. Caller must have WRITE permission for</p>
<p class="toc_element">
  <code><a href="#get">get(annotationId, x__xgafv=None)</a></code></p>
<p class="firstline">Gets an annotation. Caller must have READ permission</p>
<p class="toc_element">
  <code><a href="#search">search(body, x__xgafv=None)</a></code></p>
<p class="firstline">Searches for annotations that match the given criteria. Results are</p>
<p class="toc_element">
  <code><a href="#search_next">search_next(previous_request, previous_response)</a></code></p>
<p class="firstline">Retrieves the next page of results.</p>
<p class="toc_element">
  <code><a href="#update">update(annotationId, body, updateMask=None, x__xgafv=None)</a></code></p>
<p class="firstline">Updates an annotation. Caller must have</p>
<h3>Method Details</h3>
<div class="method">
    <code class="details" id="batchCreate">batchCreate(body, x__xgafv=None)</code>
  <pre>Creates one or more new annotations atomically. All annotations must
belong to the same annotation set. Caller must have WRITE
permission for this annotation set. For optimal performance, batch
positionally adjacent annotations together.

If the request has a systemic issue, such as an attempt to write to
an inaccessible annotation set, the entire RPC will fail accordingly. For
lesser data issues, when possible an error will be isolated to the
corresponding batch entry in the response; the remaining well formed
annotations will be created normally.

For details on the requirements for each individual annotation resource,
see
CreateAnnotation.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{
    "requestId": "A String", # A unique request ID which enables the server to detect duplicated requests.
        # If provided, duplicated requests will result in the same response; if not
        # provided, duplicated requests may result in duplicated data. For a given
        # annotation set, callers should not reuse `request_id`s when writing
        # different batches of annotations - behavior in this case is undefined.
        # A common approach is to use a UUID. For batch jobs where worker crashes are
        # a possibility, consider using some unique variant of a worker or run ID.
    "annotations": [ # The annotations to be created. At most 4096 can be specified in a single
        # request.
      { # An annotation describes a region of reference genome. The value of an
          # annotation may be one of several canonical types, supplemented by arbitrary
          # info tags. An annotation is not inherently associated with a specific
          # sample or individual (though a client could choose to use annotations in
          # this way). Example canonical annotation types are `GENE` and
          # `VARIANT`.
        "info": { # A map of additional read alignment information. This must be of the form
            # map<string, string[]> (string key mapping to a list of string values).
          "a_key": [
            "",
          ],
        },
        "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
        "end": "A String", # The end position of the range on the reference, 0-based exclusive.
        "name": "A String", # The display name of this annotation.
        "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
            # the reference genome may be transcribed as RNA. An alternative splicing
            # pattern would be represented as a separate transcript object. This field
            # is only set for annotations of type `TRANSCRIPT`.
            # reference genome may be transcribed as RNA.
          "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
              # the exact ranges of coding sequence, intersect this range with those of the
              # exons, if any. If there are any
              # exons, the
              # codingSequence must start
              # and end within them.
              #
              # Note that in some cases, the reference genome will not exactly match the
              # observed mRNA transcript e.g. due to variance in the source genome from
              # reference. In these cases,
              # exon.frame will not necessarily
              # match the expected reference reading frame and coding exon reference bases
              # cannot necessarily be concatenated to produce the original transcript mRNA.
            "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
                # 0-based inclusive. Note that this position is relative to the reference
                # start, and *not* the containing annotation start.
            "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
                # 0-based exclusive. Note that this position is relative to the reference
                # start, and *not* the containing annotation start.
          },
          "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
              # this transcript. This field should be unset for genomes where transcript
              # splicing does not occur, for example prokaryotes.
              #
              # Introns are regions of the transcript that are not included in the
              # spliced RNA product. Though not explicitly modeled here, intron ranges can
              # be deduced; all regions of this transcript that are not exons are introns.
              #
              # Exonic sequences do not necessarily code for a translational product
              # (amino acids). Only the regions of exons bounded by the
              # codingSequence correspond
              # to coding DNA sequence.
              #
              # Exons are ordered by start position and may not overlap.
            {
              "start": "A String", # The start position of the exon on this annotation's reference sequence,
                  # 0-based inclusive. Note that this is relative to the reference start, and
                  # **not** the containing annotation start.
              "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
                  # the offset of the first coding base of the exon within the reading frame
                  # of the coding DNA sequence, if any. This field is dependent on the
                  # strandedness of this annotation (see
                  # Annotation.reverse_strand).
                  # For forward stranded annotations, this offset is relative to the
                  # exon.start. For reverse
                  # strand annotations, this offset is relative to the
                  # exon.end `- 1`.
                  #
                  # Unset if this exon does not intersect the coding sequence. Upon creation
                  # of a transcript, the frame must be populated for all or none of the
                  # coding exons.
              "end": "A String", # The end position of the exon on this annotation's reference sequence,
                  # 0-based exclusive. Note that this is relative to the reference start, and
                  # *not* the containing annotation start.
            },
          ],
          "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
        },
        "variant": { # A variant annotation, which describes the effect of a variant on the
            # genome, the coding sequence, and/or higher level consequences at the
            # organism level e.g. pathogenicity. This field is only set for annotations
            # of type `VARIANT`.
          "type": "A String", # Type has been adapted from ClinVar's list of variant types.
          "effect": "A String", # Effect of the variant on the coding sequence.
          "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
              # should be provided when the variant is created.
            "A String",
          ],
          "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
              # exist at this location, create a separate variant for each one, as they
              # may represent distinct conditions.
          "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
              # It is adapted from the ClinVar controlled vocabulary for clinical
              # significance described at:
              # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
          "conditions": [ # The set of conditions associated with this variant.
              # A condition describes the way a variant influences human health.
            {
              "externalIds": [ # The set of external IDs for this condition.
                {
                  "sourceName": "A String", # The name of the source of this data.
                  "id": "A String", # The id used by the source of this data.
                },
              ],
              "conceptId": "A String", # The MedGen concept id associated with this gene.
                  # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
              "omimId": "A String", # The OMIM id for this condition.
                  # Search for these IDs at http://omim.org/
              "names": [ # A set of names for the condition.
                "A String",
              ],
            },
          ],
          "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
              # be provided when the variant is created.
        },
        "start": "A String", # The start position of the range on the reference, 0-based inclusive.
        "annotationSetId": "A String", # The annotation set to which this annotation belongs.
        "referenceName": "A String", # The display name corresponding to the reference specified by
            # `referenceId`, for example `chr1`, `1`, or `chrX`.
        "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
            # strand. Note that regardless of this field, the start/end position of the
            # range always refer to the forward strand.
        "type": "A String", # The data type for this annotation. Must match the containing annotation
            # set's type.
        "id": "A String", # The server-generated annotation ID, unique across all annotations.
      },
    ],
  }

  x__xgafv: string, V1 error format.
    Allowed values
      1 - v1 error format
      2 - v2 error format

Returns:
  An object of the form:

    {
    "entries": [ # The resulting per-annotation entries, ordered consistently with the
        # original request.
      {
        "status": { # The `Status` type defines a logical error model that is suitable for different # The creation status.
            # programming environments, including REST APIs and RPC APIs. It is used by
            # [gRPC](https://github.com/grpc). The error model is designed to be:
            #
            # - Simple to use and understand for most users
            # - Flexible enough to meet unexpected needs
            #
            # # Overview
            #
            # The `Status` message contains three pieces of data: error code, error message,
            # and error details. The error code should be an enum value of
            # google.rpc.Code, but it may accept additional error codes if needed.  The
            # error message should be a developer-facing English message that helps
            # developers *understand* and *resolve* the error. If a localized user-facing
            # error message is needed, put the localized message in the error details or
            # localize it in the client. The optional error details may contain arbitrary
            # information about the error. There is a predefined set of error detail types
            # in the package `google.rpc` that can be used for common error conditions.
            #
            # # Language mapping
            #
            # The `Status` message is the logical representation of the error model, but it
            # is not necessarily the actual wire format. When the `Status` message is
            # exposed in different client libraries and different wire protocols, it can be
            # mapped differently. For example, it will likely be mapped to some exceptions
            # in Java, but more likely mapped to some error codes in C.
            #
            # # Other uses
            #
            # The error model and the `Status` message can be used in a variety of
            # environments, either with or without APIs, to provide a
            # consistent developer experience across different environments.
            #
            # Example uses of this error model include:
            #
            # - Partial errors. If a service needs to return partial errors to the client,
            #     it may embed the `Status` in the normal response to indicate the partial
            #     errors.
            #
            # - Workflow errors. A typical workflow has multiple steps. Each step may
            #     have a `Status` message for error reporting.
            #
            # - Batch operations. If a client uses batch request and batch response, the
            #     `Status` message should be used directly inside batch response, one for
            #     each error sub-response.
            #
            # - Asynchronous operations. If an API call embeds asynchronous operation
            #     results in its response, the status of those operations should be
            #     represented directly using the `Status` message.
            #
            # - Logging. If some API errors are stored in logs, the message `Status` could
            #     be used directly after any stripping needed for security/privacy reasons.
          "message": "A String", # A developer-facing error message, which should be in English. Any
              # user-facing error message should be localized and sent in the
              # google.rpc.Status.details field, or localized by the client.
          "code": 42, # The status code, which should be an enum value of google.rpc.Code.
          "details": [ # A list of messages that carry the error details.  There will be a
              # common set of message types for APIs to use.
            {
              "a_key": "", # Properties of the object. Contains field @type with type URL.
            },
          ],
        },
        "annotation": { # An annotation describes a region of reference genome. The value of an # The created annotation, if creation was successful.
            # annotation may be one of several canonical types, supplemented by arbitrary
            # info tags. An annotation is not inherently associated with a specific
            # sample or individual (though a client could choose to use annotations in
            # this way). Example canonical annotation types are `GENE` and
            # `VARIANT`.
          "info": { # A map of additional read alignment information. This must be of the form
              # map<string, string[]> (string key mapping to a list of string values).
            "a_key": [
              "",
            ],
          },
          "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
          "end": "A String", # The end position of the range on the reference, 0-based exclusive.
          "name": "A String", # The display name of this annotation.
          "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
              # the reference genome may be transcribed as RNA. An alternative splicing
              # pattern would be represented as a separate transcript object. This field
              # is only set for annotations of type `TRANSCRIPT`.
              # reference genome may be transcribed as RNA.
            "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
                # the exact ranges of coding sequence, intersect this range with those of the
                # exons, if any. If there are any
                # exons, the
                # codingSequence must start
                # and end within them.
                #
                # Note that in some cases, the reference genome will not exactly match the
                # observed mRNA transcript e.g. due to variance in the source genome from
                # reference. In these cases,
                # exon.frame will not necessarily
                # match the expected reference reading frame and coding exon reference bases
                # cannot necessarily be concatenated to produce the original transcript mRNA.
              "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
                  # 0-based inclusive. Note that this position is relative to the reference
                  # start, and *not* the containing annotation start.
              "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
                  # 0-based exclusive. Note that this position is relative to the reference
                  # start, and *not* the containing annotation start.
            },
            "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
                # this transcript. This field should be unset for genomes where transcript
                # splicing does not occur, for example prokaryotes.
                #
                # Introns are regions of the transcript that are not included in the
                # spliced RNA product. Though not explicitly modeled here, intron ranges can
                # be deduced; all regions of this transcript that are not exons are introns.
                #
                # Exonic sequences do not necessarily code for a translational product
                # (amino acids). Only the regions of exons bounded by the
                # codingSequence correspond
                # to coding DNA sequence.
                #
                # Exons are ordered by start position and may not overlap.
              {
                "start": "A String", # The start position of the exon on this annotation's reference sequence,
                    # 0-based inclusive. Note that this is relative to the reference start, and
                    # **not** the containing annotation start.
                "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
                    # the offset of the first coding base of the exon within the reading frame
                    # of the coding DNA sequence, if any. This field is dependent on the
                    # strandedness of this annotation (see
                    # Annotation.reverse_strand).
                    # For forward stranded annotations, this offset is relative to the
                    # exon.start. For reverse
                    # strand annotations, this offset is relative to the
                    # exon.end `- 1`.
                    #
                    # Unset if this exon does not intersect the coding sequence. Upon creation
                    # of a transcript, the frame must be populated for all or none of the
                    # coding exons.
                "end": "A String", # The end position of the exon on this annotation's reference sequence,
                    # 0-based exclusive. Note that this is relative to the reference start, and
                    # *not* the containing annotation start.
              },
            ],
            "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
          },
          "variant": { # A variant annotation, which describes the effect of a variant on the
              # genome, the coding sequence, and/or higher level consequences at the
              # organism level e.g. pathogenicity. This field is only set for annotations
              # of type `VARIANT`.
            "type": "A String", # Type has been adapted from ClinVar's list of variant types.
            "effect": "A String", # Effect of the variant on the coding sequence.
            "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
                # should be provided when the variant is created.
              "A String",
            ],
            "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
                # exist at this location, create a separate variant for each one, as they
                # may represent distinct conditions.
            "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
                # It is adapted from the ClinVar controlled vocabulary for clinical
                # significance described at:
                # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
            "conditions": [ # The set of conditions associated with this variant.
                # A condition describes the way a variant influences human health.
              {
                "externalIds": [ # The set of external IDs for this condition.
                  {
                    "sourceName": "A String", # The name of the source of this data.
                    "id": "A String", # The id used by the source of this data.
                  },
                ],
                "conceptId": "A String", # The MedGen concept id associated with this gene.
                    # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
                "omimId": "A String", # The OMIM id for this condition.
                    # Search for these IDs at http://omim.org/
                "names": [ # A set of names for the condition.
                  "A String",
                ],
              },
            ],
            "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
                # be provided when the variant is created.
          },
          "start": "A String", # The start position of the range on the reference, 0-based inclusive.
          "annotationSetId": "A String", # The annotation set to which this annotation belongs.
          "referenceName": "A String", # The display name corresponding to the reference specified by
              # `referenceId`, for example `chr1`, `1`, or `chrX`.
          "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
              # strand. Note that regardless of this field, the start/end position of the
              # range always refer to the forward strand.
          "type": "A String", # The data type for this annotation. Must match the containing annotation
              # set's type.
          "id": "A String", # The server-generated annotation ID, unique across all annotations.
        },
      },
    ],
  }</pre>
</div>

<div class="method">
    <code class="details" id="create">create(body, x__xgafv=None)</code>
  <pre>Creates a new annotation. Caller must have WRITE permission
for the associated annotation set.

The following fields are required:

* annotationSetId
* referenceName or
  referenceId

### Transcripts

For annotations of type TRANSCRIPT, the following fields of
transcript must be provided:

* exons.start
* exons.end

All other fields may be optionally specified, unless documented as being
server-generated (for example, the `id` field). The annotated
range must be no longer than 100Mbp (mega base pairs). See the
Annotation resource
for additional restrictions on each field.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{ # An annotation describes a region of reference genome. The value of an
    # annotation may be one of several canonical types, supplemented by arbitrary
    # info tags. An annotation is not inherently associated with a specific
    # sample or individual (though a client could choose to use annotations in
    # this way). Example canonical annotation types are `GENE` and
    # `VARIANT`.
  "info": { # A map of additional read alignment information. This must be of the form
      # map<string, string[]> (string key mapping to a list of string values).
    "a_key": [
      "",
    ],
  },
  "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
  "end": "A String", # The end position of the range on the reference, 0-based exclusive.
  "name": "A String", # The display name of this annotation.
  "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
      # the reference genome may be transcribed as RNA. An alternative splicing
      # pattern would be represented as a separate transcript object. This field
      # is only set for annotations of type `TRANSCRIPT`.
      # reference genome may be transcribed as RNA.
    "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
        # the exact ranges of coding sequence, intersect this range with those of the
        # exons, if any. If there are any
        # exons, the
        # codingSequence must start
        # and end within them.
        #
        # Note that in some cases, the reference genome will not exactly match the
        # observed mRNA transcript e.g. due to variance in the source genome from
        # reference. In these cases,
        # exon.frame will not necessarily
        # match the expected reference reading frame and coding exon reference bases
        # cannot necessarily be concatenated to produce the original transcript mRNA.
      "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
          # 0-based inclusive. Note that this position is relative to the reference
          # start, and *not* the containing annotation start.
      "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
          # 0-based exclusive. Note that this position is relative to the reference
          # start, and *not* the containing annotation start.
    },
    "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
        # this transcript. This field should be unset for genomes where transcript
        # splicing does not occur, for example prokaryotes.
        #
        # Introns are regions of the transcript that are not included in the
        # spliced RNA product. Though not explicitly modeled here, intron ranges can
        # be deduced; all regions of this transcript that are not exons are introns.
        #
        # Exonic sequences do not necessarily code for a translational product
        # (amino acids). Only the regions of exons bounded by the
        # codingSequence correspond
        # to coding DNA sequence.
        #
        # Exons are ordered by start position and may not overlap.
      {
        "start": "A String", # The start position of the exon on this annotation's reference sequence,
            # 0-based inclusive. Note that this is relative to the reference start, and
            # **not** the containing annotation start.
        "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
            # the offset of the first coding base of the exon within the reading frame
            # of the coding DNA sequence, if any. This field is dependent on the
            # strandedness of this annotation (see
            # Annotation.reverse_strand).
            # For forward stranded annotations, this offset is relative to the
            # exon.start. For reverse
            # strand annotations, this offset is relative to the
            # exon.end `- 1`.
            #
            # Unset if this exon does not intersect the coding sequence. Upon creation
            # of a transcript, the frame must be populated for all or none of the
            # coding exons.
        "end": "A String", # The end position of the exon on this annotation's reference sequence,
            # 0-based exclusive. Note that this is relative to the reference start, and
            # *not* the containing annotation start.
      },
    ],
    "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
  },
  "variant": { # A variant annotation, which describes the effect of a variant on the
      # genome, the coding sequence, and/or higher level consequences at the
      # organism level e.g. pathogenicity. This field is only set for annotations
      # of type `VARIANT`.
    "type": "A String", # Type has been adapted from ClinVar's list of variant types.
    "effect": "A String", # Effect of the variant on the coding sequence.
    "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
        # should be provided when the variant is created.
      "A String",
    ],
    "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
        # exist at this location, create a separate variant for each one, as they
        # may represent distinct conditions.
    "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
        # It is adapted from the ClinVar controlled vocabulary for clinical
        # significance described at:
        # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
    "conditions": [ # The set of conditions associated with this variant.
        # A condition describes the way a variant influences human health.
      {
        "externalIds": [ # The set of external IDs for this condition.
          {
            "sourceName": "A String", # The name of the source of this data.
            "id": "A String", # The id used by the source of this data.
          },
        ],
        "conceptId": "A String", # The MedGen concept id associated with this gene.
            # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
        "omimId": "A String", # The OMIM id for this condition.
            # Search for these IDs at http://omim.org/
        "names": [ # A set of names for the condition.
          "A String",
        ],
      },
    ],
    "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
        # be provided when the variant is created.
  },
  "start": "A String", # The start position of the range on the reference, 0-based inclusive.
  "annotationSetId": "A String", # The annotation set to which this annotation belongs.
  "referenceName": "A String", # The display name corresponding to the reference specified by
      # `referenceId`, for example `chr1`, `1`, or `chrX`.
  "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
      # strand. Note that regardless of this field, the start/end position of the
      # range always refer to the forward strand.
  "type": "A String", # The data type for this annotation. Must match the containing annotation
      # set's type.
  "id": "A String", # The server-generated annotation ID, unique across all annotations.
}

  x__xgafv: string, V1 error format.
    Allowed values
      1 - v1 error format
      2 - v2 error format

Returns:
  An object of the form:

    { # An annotation describes a region of reference genome. The value of an
      # annotation may be one of several canonical types, supplemented by arbitrary
      # info tags. An annotation is not inherently associated with a specific
      # sample or individual (though a client could choose to use annotations in
      # this way). Example canonical annotation types are `GENE` and
      # `VARIANT`.
    "info": { # A map of additional read alignment information. This must be of the form
        # map<string, string[]> (string key mapping to a list of string values).
      "a_key": [
        "",
      ],
    },
    "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
    "end": "A String", # The end position of the range on the reference, 0-based exclusive.
    "name": "A String", # The display name of this annotation.
    "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
        # the reference genome may be transcribed as RNA. An alternative splicing
        # pattern would be represented as a separate transcript object. This field
        # is only set for annotations of type `TRANSCRIPT`.
        # reference genome may be transcribed as RNA.
      "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
          # the exact ranges of coding sequence, intersect this range with those of the
          # exons, if any. If there are any
          # exons, the
          # codingSequence must start
          # and end within them.
          #
          # Note that in some cases, the reference genome will not exactly match the
          # observed mRNA transcript e.g. due to variance in the source genome from
          # reference. In these cases,
          # exon.frame will not necessarily
          # match the expected reference reading frame and coding exon reference bases
          # cannot necessarily be concatenated to produce the original transcript mRNA.
        "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
            # 0-based inclusive. Note that this position is relative to the reference
            # start, and *not* the containing annotation start.
        "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
            # 0-based exclusive. Note that this position is relative to the reference
            # start, and *not* the containing annotation start.
      },
      "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
          # this transcript. This field should be unset for genomes where transcript
          # splicing does not occur, for example prokaryotes.
          #
          # Introns are regions of the transcript that are not included in the
          # spliced RNA product. Though not explicitly modeled here, intron ranges can
          # be deduced; all regions of this transcript that are not exons are introns.
          #
          # Exonic sequences do not necessarily code for a translational product
          # (amino acids). Only the regions of exons bounded by the
          # codingSequence correspond
          # to coding DNA sequence.
          #
          # Exons are ordered by start position and may not overlap.
        {
          "start": "A String", # The start position of the exon on this annotation's reference sequence,
              # 0-based inclusive. Note that this is relative to the reference start, and
              # **not** the containing annotation start.
          "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
              # the offset of the first coding base of the exon within the reading frame
              # of the coding DNA sequence, if any. This field is dependent on the
              # strandedness of this annotation (see
              # Annotation.reverse_strand).
              # For forward stranded annotations, this offset is relative to the
              # exon.start. For reverse
              # strand annotations, this offset is relative to the
              # exon.end `- 1`.
              #
              # Unset if this exon does not intersect the coding sequence. Upon creation
              # of a transcript, the frame must be populated for all or none of the
              # coding exons.
          "end": "A String", # The end position of the exon on this annotation's reference sequence,
              # 0-based exclusive. Note that this is relative to the reference start, and
              # *not* the containing annotation start.
        },
      ],
      "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
    },
    "variant": { # A variant annotation, which describes the effect of a variant on the
        # genome, the coding sequence, and/or higher level consequences at the
        # organism level e.g. pathogenicity. This field is only set for annotations
        # of type `VARIANT`.
      "type": "A String", # Type has been adapted from ClinVar's list of variant types.
      "effect": "A String", # Effect of the variant on the coding sequence.
      "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
          # should be provided when the variant is created.
        "A String",
      ],
      "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
          # exist at this location, create a separate variant for each one, as they
          # may represent distinct conditions.
      "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
          # It is adapted from the ClinVar controlled vocabulary for clinical
          # significance described at:
          # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
      "conditions": [ # The set of conditions associated with this variant.
          # A condition describes the way a variant influences human health.
        {
          "externalIds": [ # The set of external IDs for this condition.
            {
              "sourceName": "A String", # The name of the source of this data.
              "id": "A String", # The id used by the source of this data.
            },
          ],
          "conceptId": "A String", # The MedGen concept id associated with this gene.
              # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
          "omimId": "A String", # The OMIM id for this condition.
              # Search for these IDs at http://omim.org/
          "names": [ # A set of names for the condition.
            "A String",
          ],
        },
      ],
      "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
          # be provided when the variant is created.
    },
    "start": "A String", # The start position of the range on the reference, 0-based inclusive.
    "annotationSetId": "A String", # The annotation set to which this annotation belongs.
    "referenceName": "A String", # The display name corresponding to the reference specified by
        # `referenceId`, for example `chr1`, `1`, or `chrX`.
    "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
        # strand. Note that regardless of this field, the start/end position of the
        # range always refer to the forward strand.
    "type": "A String", # The data type for this annotation. Must match the containing annotation
        # set's type.
    "id": "A String", # The server-generated annotation ID, unique across all annotations.
  }</pre>
</div>

<div class="method">
    <code class="details" id="delete">delete(annotationId, x__xgafv=None)</code>
  <pre>Deletes an annotation. Caller must have WRITE permission for
the associated annotation set.

Args:
  annotationId: string, The ID of the annotation to be deleted. (required)
  x__xgafv: string, V1 error format.
    Allowed values
      1 - v1 error format
      2 - v2 error format

Returns:
  An object of the form:

    { # A generic empty message that you can re-use to avoid defining duplicated
      # empty messages in your APIs. A typical example is to use it as the request
      # or the response type of an API method. For instance:
      #
      #     service Foo {
      #       rpc Bar(google.protobuf.Empty) returns (google.protobuf.Empty);
      #     }
      #
      # The JSON representation for `Empty` is empty JSON object `{}`.
  }</pre>
</div>

<div class="method">
    <code class="details" id="get">get(annotationId, x__xgafv=None)</code>
  <pre>Gets an annotation. Caller must have READ permission
for the associated annotation set.

Args:
  annotationId: string, The ID of the annotation to be retrieved. (required)
  x__xgafv: string, V1 error format.
    Allowed values
      1 - v1 error format
      2 - v2 error format

Returns:
  An object of the form:

    { # An annotation describes a region of reference genome. The value of an
      # annotation may be one of several canonical types, supplemented by arbitrary
      # info tags. An annotation is not inherently associated with a specific
      # sample or individual (though a client could choose to use annotations in
      # this way). Example canonical annotation types are `GENE` and
      # `VARIANT`.
    "info": { # A map of additional read alignment information. This must be of the form
        # map<string, string[]> (string key mapping to a list of string values).
      "a_key": [
        "",
      ],
    },
    "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
    "end": "A String", # The end position of the range on the reference, 0-based exclusive.
    "name": "A String", # The display name of this annotation.
    "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
        # the reference genome may be transcribed as RNA. An alternative splicing
        # pattern would be represented as a separate transcript object. This field
        # is only set for annotations of type `TRANSCRIPT`.
        # reference genome may be transcribed as RNA.
      "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
          # the exact ranges of coding sequence, intersect this range with those of the
          # exons, if any. If there are any
          # exons, the
          # codingSequence must start
          # and end within them.
          #
          # Note that in some cases, the reference genome will not exactly match the
          # observed mRNA transcript e.g. due to variance in the source genome from
          # reference. In these cases,
          # exon.frame will not necessarily
          # match the expected reference reading frame and coding exon reference bases
          # cannot necessarily be concatenated to produce the original transcript mRNA.
        "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
            # 0-based inclusive. Note that this position is relative to the reference
            # start, and *not* the containing annotation start.
        "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
            # 0-based exclusive. Note that this position is relative to the reference
            # start, and *not* the containing annotation start.
      },
      "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
          # this transcript. This field should be unset for genomes where transcript
          # splicing does not occur, for example prokaryotes.
          #
          # Introns are regions of the transcript that are not included in the
          # spliced RNA product. Though not explicitly modeled here, intron ranges can
          # be deduced; all regions of this transcript that are not exons are introns.
          #
          # Exonic sequences do not necessarily code for a translational product
          # (amino acids). Only the regions of exons bounded by the
          # codingSequence correspond
          # to coding DNA sequence.
          #
          # Exons are ordered by start position and may not overlap.
        {
          "start": "A String", # The start position of the exon on this annotation's reference sequence,
              # 0-based inclusive. Note that this is relative to the reference start, and
              # **not** the containing annotation start.
          "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
              # the offset of the first coding base of the exon within the reading frame
              # of the coding DNA sequence, if any. This field is dependent on the
              # strandedness of this annotation (see
              # Annotation.reverse_strand).
              # For forward stranded annotations, this offset is relative to the
              # exon.start. For reverse
              # strand annotations, this offset is relative to the
              # exon.end `- 1`.
              #
              # Unset if this exon does not intersect the coding sequence. Upon creation
              # of a transcript, the frame must be populated for all or none of the
              # coding exons.
          "end": "A String", # The end position of the exon on this annotation's reference sequence,
              # 0-based exclusive. Note that this is relative to the reference start, and
              # *not* the containing annotation start.
        },
      ],
      "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
    },
    "variant": { # A variant annotation, which describes the effect of a variant on the
        # genome, the coding sequence, and/or higher level consequences at the
        # organism level e.g. pathogenicity. This field is only set for annotations
        # of type `VARIANT`.
      "type": "A String", # Type has been adapted from ClinVar's list of variant types.
      "effect": "A String", # Effect of the variant on the coding sequence.
      "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
          # should be provided when the variant is created.
        "A String",
      ],
      "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
          # exist at this location, create a separate variant for each one, as they
          # may represent distinct conditions.
      "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
          # It is adapted from the ClinVar controlled vocabulary for clinical
          # significance described at:
          # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
      "conditions": [ # The set of conditions associated with this variant.
          # A condition describes the way a variant influences human health.
        {
          "externalIds": [ # The set of external IDs for this condition.
            {
              "sourceName": "A String", # The name of the source of this data.
              "id": "A String", # The id used by the source of this data.
            },
          ],
          "conceptId": "A String", # The MedGen concept id associated with this gene.
              # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
          "omimId": "A String", # The OMIM id for this condition.
              # Search for these IDs at http://omim.org/
          "names": [ # A set of names for the condition.
            "A String",
          ],
        },
      ],
      "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
          # be provided when the variant is created.
    },
    "start": "A String", # The start position of the range on the reference, 0-based inclusive.
    "annotationSetId": "A String", # The annotation set to which this annotation belongs.
    "referenceName": "A String", # The display name corresponding to the reference specified by
        # `referenceId`, for example `chr1`, `1`, or `chrX`.
    "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
        # strand. Note that regardless of this field, the start/end position of the
        # range always refer to the forward strand.
    "type": "A String", # The data type for this annotation. Must match the containing annotation
        # set's type.
    "id": "A String", # The server-generated annotation ID, unique across all annotations.
  }</pre>
</div>

<div class="method">
    <code class="details" id="search">search(body, x__xgafv=None)</code>
  <pre>Searches for annotations that match the given criteria. Results are
ordered by genomic coordinate (by reference sequence, then position).
Annotations with equivalent genomic coordinates are returned in an
unspecified order. This order is consistent, such that two queries for the
same content (regardless of page size) yield annotations in the same order
across their respective streams of paginated responses. Caller must have
READ permission for the queried annotation sets.

Args:
  body: object, The request body. (required)
    The object takes the form of:

{
    "referenceId": "A String", # The ID of the reference to query.
    "end": "A String", # The end position of the range on the reference, 0-based exclusive. If
        # referenceId or
        # referenceName
        # must be specified, Defaults to the length of the reference.
    "pageSize": 42, # The maximum number of results to return in a single page. If unspecified,
        # defaults to 256. The maximum value is 2048.
    "start": "A String", # The start position of the range on the reference, 0-based inclusive. If
        # specified,
        # referenceId or
        # referenceName
        # must be specified. Defaults to 0.
    "annotationSetIds": [ # Required. The annotation sets to search within. The caller must have
        # `READ` access to these annotation sets.
        # All queried annotation sets must have the same type.
      "A String",
    ],
    "pageToken": "A String", # The continuation token, which is used to page through large result sets.
        # To get the next page of results, set this parameter to the value of
        # `nextPageToken` from the previous response.
    "referenceName": "A String", # The name of the reference to query, within the reference set associated
        # with this query.
  }

  x__xgafv: string, V1 error format.
    Allowed values
      1 - v1 error format
      2 - v2 error format

Returns:
  An object of the form:

    {
    "nextPageToken": "A String", # The continuation token, which is used to page through large result sets.
        # Provide this value in a subsequent request to return the next page of
        # results. This field will be empty if there aren't any additional results.
    "annotations": [ # The matching annotations.
      { # An annotation describes a region of reference genome. The value of an
          # annotation may be one of several canonical types, supplemented by arbitrary
          # info tags. An annotation is not inherently associated with a specific
          # sample or individual (though a client could choose to use annotations in
          # this way). Example canonical annotation types are `GENE` and
          # `VARIANT`.
        "info": { # A map of additional read alignment information. This must be of the form
            # map<string, string[]> (string key mapping to a list of string values).
          "a_key": [
            "",
          ],
        },
        "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
        "end": "A String", # The end position of the range on the reference, 0-based exclusive.
        "name": "A String", # The display name of this annotation.
        "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
            # the reference genome may be transcribed as RNA. An alternative splicing
            # pattern would be represented as a separate transcript object. This field
            # is only set for annotations of type `TRANSCRIPT`.
            # reference genome may be transcribed as RNA.
          "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
              # the exact ranges of coding sequence, intersect this range with those of the
              # exons, if any. If there are any
              # exons, the
              # codingSequence must start
              # and end within them.
              #
              # Note that in some cases, the reference genome will not exactly match the
              # observed mRNA transcript e.g. due to variance in the source genome from
              # reference. In these cases,
              # exon.frame will not necessarily
              # match the expected reference reading frame and coding exon reference bases
              # cannot necessarily be concatenated to produce the original transcript mRNA.
            "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
                # 0-based inclusive. Note that this position is relative to the reference
                # start, and *not* the containing annotation start.
            "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
                # 0-based exclusive. Note that this position is relative to the reference
                # start, and *not* the containing annotation start.
          },
          "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
              # this transcript. This field should be unset for genomes where transcript
              # splicing does not occur, for example prokaryotes.
              #
              # Introns are regions of the transcript that are not included in the
              # spliced RNA product. Though not explicitly modeled here, intron ranges can
              # be deduced; all regions of this transcript that are not exons are introns.
              #
              # Exonic sequences do not necessarily code for a translational product
              # (amino acids). Only the regions of exons bounded by the
              # codingSequence correspond
              # to coding DNA sequence.
              #
              # Exons are ordered by start position and may not overlap.
            {
              "start": "A String", # The start position of the exon on this annotation's reference sequence,
                  # 0-based inclusive. Note that this is relative to the reference start, and
                  # **not** the containing annotation start.
              "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
                  # the offset of the first coding base of the exon within the reading frame
                  # of the coding DNA sequence, if any. This field is dependent on the
                  # strandedness of this annotation (see
                  # Annotation.reverse_strand).
                  # For forward stranded annotations, this offset is relative to the
                  # exon.start. For reverse
                  # strand annotations, this offset is relative to the
                  # exon.end `- 1`.
                  #
                  # Unset if this exon does not intersect the coding sequence. Upon creation
                  # of a transcript, the frame must be populated for all or none of the
                  # coding exons.
              "end": "A String", # The end position of the exon on this annotation's reference sequence,
                  # 0-based exclusive. Note that this is relative to the reference start, and
                  # *not* the containing annotation start.
            },
          ],
          "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
        },
        "variant": { # A variant annotation, which describes the effect of a variant on the
            # genome, the coding sequence, and/or higher level consequences at the
            # organism level e.g. pathogenicity. This field is only set for annotations
            # of type `VARIANT`.
          "type": "A String", # Type has been adapted from ClinVar's list of variant types.
          "effect": "A String", # Effect of the variant on the coding sequence.
          "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
              # should be provided when the variant is created.
            "A String",
          ],
          "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
              # exist at this location, create a separate variant for each one, as they
              # may represent distinct conditions.
          "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
              # It is adapted from the ClinVar controlled vocabulary for clinical
              # significance described at:
              # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
          "conditions": [ # The set of conditions associated with this variant.
              # A condition describes the way a variant influences human health.
            {
              "externalIds": [ # The set of external IDs for this condition.
                {
                  "sourceName": "A String", # The name of the source of this data.
                  "id": "A String", # The id used by the source of this data.
                },
              ],
              "conceptId": "A String", # The MedGen concept id associated with this gene.
                  # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
              "omimId": "A String", # The OMIM id for this condition.
                  # Search for these IDs at http://omim.org/
              "names": [ # A set of names for the condition.
                "A String",
              ],
            },
          ],
          "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
              # be provided when the variant is created.
        },
        "start": "A String", # The start position of the range on the reference, 0-based inclusive.
        "annotationSetId": "A String", # The annotation set to which this annotation belongs.
        "referenceName": "A String", # The display name corresponding to the reference specified by
            # `referenceId`, for example `chr1`, `1`, or `chrX`.
        "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
            # strand. Note that regardless of this field, the start/end position of the
            # range always refer to the forward strand.
        "type": "A String", # The data type for this annotation. Must match the containing annotation
            # set's type.
        "id": "A String", # The server-generated annotation ID, unique across all annotations.
      },
    ],
  }</pre>
</div>

<div class="method">
    <code class="details" id="search_next">search_next(previous_request, previous_response)</code>
  <pre>Retrieves the next page of results.

Args:
  previous_request: The request for the previous page. (required)
  previous_response: The response from the request for the previous page. (required)

Returns:
  A request object that you can call 'execute()' on to request the next
  page. Returns None if there are no more items in the collection.
    </pre>
</div>

<div class="method">
    <code class="details" id="update">update(annotationId, body, updateMask=None, x__xgafv=None)</code>
  <pre>Updates an annotation. Caller must have
WRITE permission for the associated dataset.

Args:
  annotationId: string, The ID of the annotation to be updated. (required)
  body: object, The request body. (required)
    The object takes the form of:

{ # An annotation describes a region of reference genome. The value of an
    # annotation may be one of several canonical types, supplemented by arbitrary
    # info tags. An annotation is not inherently associated with a specific
    # sample or individual (though a client could choose to use annotations in
    # this way). Example canonical annotation types are `GENE` and
    # `VARIANT`.
  "info": { # A map of additional read alignment information. This must be of the form
      # map<string, string[]> (string key mapping to a list of string values).
    "a_key": [
      "",
    ],
  },
  "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
  "end": "A String", # The end position of the range on the reference, 0-based exclusive.
  "name": "A String", # The display name of this annotation.
  "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
      # the reference genome may be transcribed as RNA. An alternative splicing
      # pattern would be represented as a separate transcript object. This field
      # is only set for annotations of type `TRANSCRIPT`.
      # reference genome may be transcribed as RNA.
    "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
        # the exact ranges of coding sequence, intersect this range with those of the
        # exons, if any. If there are any
        # exons, the
        # codingSequence must start
        # and end within them.
        #
        # Note that in some cases, the reference genome will not exactly match the
        # observed mRNA transcript e.g. due to variance in the source genome from
        # reference. In these cases,
        # exon.frame will not necessarily
        # match the expected reference reading frame and coding exon reference bases
        # cannot necessarily be concatenated to produce the original transcript mRNA.
      "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
          # 0-based inclusive. Note that this position is relative to the reference
          # start, and *not* the containing annotation start.
      "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
          # 0-based exclusive. Note that this position is relative to the reference
          # start, and *not* the containing annotation start.
    },
    "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
        # this transcript. This field should be unset for genomes where transcript
        # splicing does not occur, for example prokaryotes.
        #
        # Introns are regions of the transcript that are not included in the
        # spliced RNA product. Though not explicitly modeled here, intron ranges can
        # be deduced; all regions of this transcript that are not exons are introns.
        #
        # Exonic sequences do not necessarily code for a translational product
        # (amino acids). Only the regions of exons bounded by the
        # codingSequence correspond
        # to coding DNA sequence.
        #
        # Exons are ordered by start position and may not overlap.
      {
        "start": "A String", # The start position of the exon on this annotation's reference sequence,
            # 0-based inclusive. Note that this is relative to the reference start, and
            # **not** the containing annotation start.
        "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
            # the offset of the first coding base of the exon within the reading frame
            # of the coding DNA sequence, if any. This field is dependent on the
            # strandedness of this annotation (see
            # Annotation.reverse_strand).
            # For forward stranded annotations, this offset is relative to the
            # exon.start. For reverse
            # strand annotations, this offset is relative to the
            # exon.end `- 1`.
            #
            # Unset if this exon does not intersect the coding sequence. Upon creation
            # of a transcript, the frame must be populated for all or none of the
            # coding exons.
        "end": "A String", # The end position of the exon on this annotation's reference sequence,
            # 0-based exclusive. Note that this is relative to the reference start, and
            # *not* the containing annotation start.
      },
    ],
    "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
  },
  "variant": { # A variant annotation, which describes the effect of a variant on the
      # genome, the coding sequence, and/or higher level consequences at the
      # organism level e.g. pathogenicity. This field is only set for annotations
      # of type `VARIANT`.
    "type": "A String", # Type has been adapted from ClinVar's list of variant types.
    "effect": "A String", # Effect of the variant on the coding sequence.
    "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
        # should be provided when the variant is created.
      "A String",
    ],
    "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
        # exist at this location, create a separate variant for each one, as they
        # may represent distinct conditions.
    "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
        # It is adapted from the ClinVar controlled vocabulary for clinical
        # significance described at:
        # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
    "conditions": [ # The set of conditions associated with this variant.
        # A condition describes the way a variant influences human health.
      {
        "externalIds": [ # The set of external IDs for this condition.
          {
            "sourceName": "A String", # The name of the source of this data.
            "id": "A String", # The id used by the source of this data.
          },
        ],
        "conceptId": "A String", # The MedGen concept id associated with this gene.
            # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
        "omimId": "A String", # The OMIM id for this condition.
            # Search for these IDs at http://omim.org/
        "names": [ # A set of names for the condition.
          "A String",
        ],
      },
    ],
    "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
        # be provided when the variant is created.
  },
  "start": "A String", # The start position of the range on the reference, 0-based inclusive.
  "annotationSetId": "A String", # The annotation set to which this annotation belongs.
  "referenceName": "A String", # The display name corresponding to the reference specified by
      # `referenceId`, for example `chr1`, `1`, or `chrX`.
  "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
      # strand. Note that regardless of this field, the start/end position of the
      # range always refer to the forward strand.
  "type": "A String", # The data type for this annotation. Must match the containing annotation
      # set's type.
  "id": "A String", # The server-generated annotation ID, unique across all annotations.
}

  updateMask: string, An optional mask specifying which fields to update. Mutable fields are
name,
variant,
transcript, and
info. If unspecified, all mutable
fields will be updated.
  x__xgafv: string, V1 error format.
    Allowed values
      1 - v1 error format
      2 - v2 error format

Returns:
  An object of the form:

    { # An annotation describes a region of reference genome. The value of an
      # annotation may be one of several canonical types, supplemented by arbitrary
      # info tags. An annotation is not inherently associated with a specific
      # sample or individual (though a client could choose to use annotations in
      # this way). Example canonical annotation types are `GENE` and
      # `VARIANT`.
    "info": { # A map of additional read alignment information. This must be of the form
        # map<string, string[]> (string key mapping to a list of string values).
      "a_key": [
        "",
      ],
    },
    "referenceId": "A String", # The ID of the Google Genomics reference associated with this range.
    "end": "A String", # The end position of the range on the reference, 0-based exclusive.
    "name": "A String", # The display name of this annotation.
    "transcript": { # A transcript represents the assertion that a particular region of the # A transcript value represents the assertion that a particular region of
        # the reference genome may be transcribed as RNA. An alternative splicing
        # pattern would be represented as a separate transcript object. This field
        # is only set for annotations of type `TRANSCRIPT`.
        # reference genome may be transcribed as RNA.
      "codingSequence": { # The range of the coding sequence for this transcript, if any. To determine
          # the exact ranges of coding sequence, intersect this range with those of the
          # exons, if any. If there are any
          # exons, the
          # codingSequence must start
          # and end within them.
          #
          # Note that in some cases, the reference genome will not exactly match the
          # observed mRNA transcript e.g. due to variance in the source genome from
          # reference. In these cases,
          # exon.frame will not necessarily
          # match the expected reference reading frame and coding exon reference bases
          # cannot necessarily be concatenated to produce the original transcript mRNA.
        "start": "A String", # The start of the coding sequence on this annotation's reference sequence,
            # 0-based inclusive. Note that this position is relative to the reference
            # start, and *not* the containing annotation start.
        "end": "A String", # The end of the coding sequence on this annotation's reference sequence,
            # 0-based exclusive. Note that this position is relative to the reference
            # start, and *not* the containing annotation start.
      },
      "exons": [ # The <a href="http://en.wikipedia.org/wiki/Exon">exons</a> that compose
          # this transcript. This field should be unset for genomes where transcript
          # splicing does not occur, for example prokaryotes.
          #
          # Introns are regions of the transcript that are not included in the
          # spliced RNA product. Though not explicitly modeled here, intron ranges can
          # be deduced; all regions of this transcript that are not exons are introns.
          #
          # Exonic sequences do not necessarily code for a translational product
          # (amino acids). Only the regions of exons bounded by the
          # codingSequence correspond
          # to coding DNA sequence.
          #
          # Exons are ordered by start position and may not overlap.
        {
          "start": "A String", # The start position of the exon on this annotation's reference sequence,
              # 0-based inclusive. Note that this is relative to the reference start, and
              # **not** the containing annotation start.
          "frame": 42, # The frame of this exon. Contains a value of 0, 1, or 2, which indicates
              # the offset of the first coding base of the exon within the reading frame
              # of the coding DNA sequence, if any. This field is dependent on the
              # strandedness of this annotation (see
              # Annotation.reverse_strand).
              # For forward stranded annotations, this offset is relative to the
              # exon.start. For reverse
              # strand annotations, this offset is relative to the
              # exon.end `- 1`.
              #
              # Unset if this exon does not intersect the coding sequence. Upon creation
              # of a transcript, the frame must be populated for all or none of the
              # coding exons.
          "end": "A String", # The end position of the exon on this annotation's reference sequence,
              # 0-based exclusive. Note that this is relative to the reference start, and
              # *not* the containing annotation start.
        },
      ],
      "geneId": "A String", # The annotation ID of the gene from which this transcript is transcribed.
    },
    "variant": { # A variant annotation, which describes the effect of a variant on the
        # genome, the coding sequence, and/or higher level consequences at the
        # organism level e.g. pathogenicity. This field is only set for annotations
        # of type `VARIANT`.
      "type": "A String", # Type has been adapted from ClinVar's list of variant types.
      "effect": "A String", # Effect of the variant on the coding sequence.
      "transcriptIds": [ # Google annotation IDs of the transcripts affected by this variant. These
          # should be provided when the variant is created.
        "A String",
      ],
      "alternateBases": "A String", # The alternate allele for this variant. If multiple alternate alleles
          # exist at this location, create a separate variant for each one, as they
          # may represent distinct conditions.
      "clinicalSignificance": "A String", # Describes the clinical significance of a variant.
          # It is adapted from the ClinVar controlled vocabulary for clinical
          # significance described at:
          # http://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/
      "conditions": [ # The set of conditions associated with this variant.
          # A condition describes the way a variant influences human health.
        {
          "externalIds": [ # The set of external IDs for this condition.
            {
              "sourceName": "A String", # The name of the source of this data.
              "id": "A String", # The id used by the source of this data.
            },
          ],
          "conceptId": "A String", # The MedGen concept id associated with this gene.
              # Search for these IDs at http://www.ncbi.nlm.nih.gov/medgen/
          "omimId": "A String", # The OMIM id for this condition.
              # Search for these IDs at http://omim.org/
          "names": [ # A set of names for the condition.
            "A String",
          ],
        },
      ],
      "geneId": "A String", # Google annotation ID of the gene affected by this variant. This should
          # be provided when the variant is created.
    },
    "start": "A String", # The start position of the range on the reference, 0-based inclusive.
    "annotationSetId": "A String", # The annotation set to which this annotation belongs.
    "referenceName": "A String", # The display name corresponding to the reference specified by
        # `referenceId`, for example `chr1`, `1`, or `chrX`.
    "reverseStrand": True or False, # Whether this range refers to the reverse strand, as opposed to the forward
        # strand. Note that regardless of this field, the start/end position of the
        # range always refer to the forward strand.
    "type": "A String", # The data type for this annotation. Must match the containing annotation
        # set's type.
    "id": "A String", # The server-generated annotation ID, unique across all annotations.
  }</pre>
</div>

</body></html>